libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -462,9 +476,12 @@ internal class MapboxTripSession(
     /**
      * Sensor event consumed by native
      */
-    override fun updateSensorEvent(sensorEvent: SensorEvent) {
-        SensorMapper.toSensorData(sensorEvent, logger)?.let { sensorData ->
-            navigator.updateSensorData(sensorData)
+    override fun updateSensorEvent(sensorEvent: SensorEvent, callback: SensorEventUpdatedCallback) {
+        mainJobController.scope.launch {
+            SensorMapper.toSensorData(sensorEvent, logger)?.let { sensorData ->
+                val result = navigator.updateSensorData(sensorData)
+                callback.onSensorEventUpdated(result)
```

Comment:
Should we put this in a `finally` block?

Reply:
And what if `toSensorData` returned `null`?

Reply:
you are right, fixed